src/install/lock.ts

```diff
@@ -258,14 +256,18 @@ export async function extractLockConstraintsAndMap (map: IImportMap, preloadUrls
       const pkgUrl = parsedTarget ? resolver.pkgToUrl(parsedTarget.pkg, parsedTarget.source) : await resolver.getPackageBase(targetUrl);
       const subpath = '.' + targetUrl.slice(pkgUrl.length - 1) as '.' | `./{string}`;
       pkgUrls.add(pkgUrl);
-      const exportSubpath = parsedTarget && await resolver.getExportResolution(pkgUrl, subpath, key);
-      if (exportSubpath) {
+      const exportSubpath = await resolver.getExportResolution(pkgUrl, subpath, key);
+
+      // If the plain specifier resolves to a package on some provider's CDN, 
+      // and we can find a suitable export for the module, then we can lock 
+      // the package down to an exact version:
+      if (parsedTarget && exportSubpath) {
```

Comment:
Maybe better to keep the existing logic unless there is a good reason to change it?